src/configure/clients/devOps/azureDevOpsClient.ts

```diff
@@ -1,35 +1,90 @@
-        return this.serviceClient.sendRequest(urlBasedRequestPrepareOptions);
+        return this.restClient.sendRequest(urlBasedRequestPrepareOptions);
     }
 
-    public async listOrganizations(): Promise<Organization[]> {
-        if (this.organizationMap || this.listOrgPromise) {
-            return this.organizationMap ? this.organizationMap : this.listOrgPromise;
-        }
+    public async createOrganization(organizationName: string): Promise<any> {
+        return this.restClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: "https://app.vsaex.visualstudio.com/_apis/HostAcquisition/collections",
+            headers: {
+                "Content-Type": "application/json"
+            },
+            method: "POST",
+            queryParameters: {
+                "collectionName": organizationName,
+                "api-version": "4.0-preview.1",
+                "preferredRegion": "CUS"
+            },
+            body: {
+                "VisualStudio.Services.HostResolution.UseCodexDomainForHostCreation": "true"
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
 
-        let listOrganizaitonResponse = await this.getConnectionData()
+    public async createProject(organizationName: string, projectName: string): Promise<any> {
+        let collectionUrl = `https://dev.azure.com/${organizationName}`;
+
+        return this.restClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: `${collectionUrl}/_apis/projects`,
+            headers: {
+                "Content-Type": "application/json"
+            },
+            method: "POST",
+            queryParameters: {
+                "api-version": "5.0"
+            },
+            body: {
+                "name": projectName,
+                "visibility": 0,
+                "capabilities": {
+                    "versioncontrol": {"sourceControlType": "Git" },
+                    "processTemplate": { "templateTypeId": "adcc42ab-9882-485e-a3ed-7678f01f66bc" }
+                }
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        })
+        .then((operation) => {
+            if(operation.url) {
+                return this.monitorOperationStatus(operation.url);
+            }
+            else {
+                throw new Error(util.format(Messages.failedToCreateAzureDevOpsProject, operation.message));
+            }
+        });
+    }
+
+    public async listOrganizations(forceRefresh?: boolean): Promise<Organization[]> {
+        if (!this.listOrgPromise || forceRefresh) {
+            this.listOrgPromise = this.getUserData()
             .then((connectionData) => {
-                return this.serviceClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+                return this.restClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
```

Comment:
Do we need to handle for MSA accounts here?

Reply:
Currently this experience does not support MSA accounts.